### PR TITLE
Initial LAPIC support 

### PR DESCRIPTION
**Pull request by StaticSaga**

### CODE REVIEW

File: src/platform/x86_64/apic.zig [hunk 1]
```diff
@@ -2,6 +2,34 @@ const os = @import("root").os;
 const std = @import("std");
 const builtin = @import("builtin");
 
+
+var lapic: *volatile [0x100]u32 = undefined;
+
+const IA32_APIC_BASE = @import("regs.zig").MSR(u64, 0x0000001B);
```
Comment (notYuriy) [inline src/platform/x86_64/apic.zig:7]: sneak_case

Comment (StaticSaga) [reply src/platform/x86_64/apic.zig:7]: CONSTANTS_ARE_IN_UPPERCASE (afaik)

Comment (N00byEdge) [reply src/platform/x86_64/apic.zig:7]: How do you feel about the constants at the bottom?

Comment (notYuriy) [reply src/platform/x86_64/apic.zig:7]: its not exactly a constant. Its MSR you write to and read from

Comment (notYuriy) [reply src/platform/x86_64/apic.zig:7]: @N00byEdge Until we get a predefined order of things in the file, its fine I guess

**Review by N00byEdge — APPROVED** [review]

File: src/platform/x86_64/apic.zig (hunk 1 above)
Comment (StaticSaga) [reply src/platform/x86_64/apic.zig]: Fixed

**Comment by notYuriy** [comment]
LGTM

File: src/platform/x86_64/apic.zig [hunk 2]
```diff
@@ -76,3 +100,11 @@ pub fn handle_madt(madt: []u8) void {
...
   }
 }
+
+const IA32_APIC_BASE = @import("regs.zig").MSR(u64, 0x0000001B);
+const LVT_TIMER = 0x320 / 4;
+const TIMER_MODE_PERIODIC = 1 << 17;
+const TIMER_DIV = 0x3E0 / 4;
+const TIMER_INITCNT = 0x380 / 4;
+const SPURIOUS = 0xF0 / 4;
+const EOI = 0xB0 / 4;
```
Comment (N00byEdge) [inline src/platform/x86_64/apic.zig:110]: NL before EOF

**Pull request merged**